src/Distortion.cpp

```diff
@@ -200,6 +200,64 @@ void removeDistortion(double dx, double dy, double &ux, double &uy,
       ux = dx + dr_x;
       uy = dy + dr_y;
     }
+    break;
+    // The dawn distortion model is "reversed" from other distortion models so
+    // the remove function iteratively computes undistorted coordinates based on
+    // the distorted coordinates, rather than iteratively computing distorted coordinates
+    // to undistorted coordinates.
+    case DAWNFC: {
+      double offsetSqrd;
+      int    numAttempts;
+      double delta;
+      bool    done;
+
+      /****************************************************************************
+      * Pre-loop intializations
+      ****************************************************************************/
+
+      numAttempts = 1;
+      delta = 0.00001;
+
+      offsetSqrd = dy * dy + dx * dx;
+      double guess_dx, guess_dy;
+      double guess_ux, guess_uy;
+
+      /****************************************************************************
+      * Loop ...
+      ****************************************************************************/
+      do {
+        guess_ux = dx / (1.0 + opticalDistCoeffs[0] * offsetSqrd);
+        guess_uy = dy / (1.0 + opticalDistCoeffs[0] * offsetSqrd);
+
+        offsetSqrd = guess_uy * guess_uy + guess_ux * guess_ux;
+
+        guess_dx = guess_ux * (1.0 + opticalDistCoeffs[0] * offsetSqrd);
+        guess_dy = guess_uy * (1.0 + opticalDistCoeffs[0] * offsetSqrd);
+
+        done = true;
+        if(abs(guess_dy - dy) > delta) {
+          done = false;
+        }
+
+        if(abs(guess_dx - dx) > delta) {
+          done = false;
+        }
```

Comment:
usually you want to check if your under tolerance and then set done to true. This is a little confusing because done will be set to true and then immediately set to false 90% of the time. You can condense this into a single && statement and it will even short circuit for a tiny savings.

Reply:
I think this was copied from ISIS, so it may need to be changed there, too?

Reply:
This is a very minor thing, I wouldn't worry about it in ISIS. It's unusual, but not incorrect.